### PR TITLE
❌ Delete broadcasts PR#5

### PR DESCRIPTION
**Pull request by susanm74**

### CODE REVIEW

File: temba/msgs/tests.py [hunk 1]
```diff
@@ -2319,6 +2319,31 @@ def test_scheduled_update(self):
         self.assertEqual(broadcast.base_language, "base")
         self.assertEqual(set(broadcast.contacts.all()), {self.frank})
 
+    def test_scheduled_deactivate(self):
```
Comment (rowanseymour) [inline temba/msgs/tests.py:2322]: just use `delete` as the verb - we consider setting `is_active=False` a soft delete

File: temba/msgs/views.py [hunk 2]
```diff
@@ -357,6 +370,32 @@ def save(self, *args, **kwargs):
             broadcast.save()
             return broadcast
 
+    class ScheduledDeactivate(ModalMixin, OrgObjPermsMixin, SmartDeleteView):
```
Comment (rowanseymour) [inline temba/msgs/views.py:373]: ScheduledDelete

File: temba/msgs/views.py [hunk 3]
```diff
@@ -357,6 +370,32 @@ def save(self, *args, **kwargs):
...
+        cancel_url = "id@msgs.broadcast_scheduled_read"
+        success_url = "@msgs.broadcast_scheduled"
+        fields = ("id",)
+        submit_button_name = _("Delete")
+        # custom object name because broadcasts don't have a name field
+        object_name = ""
+
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+            context["object_name"] = truncate(self.object.get_text(), 15)
```
Comment (rowanseymour) [inline temba/msgs/views.py:386]: don't try to give it a name - just confirm with..

```
You are about to delete this scheduled message. There is no way to undo this. Are you sure?
```

File: temba/msgs/views.py [hunk 4]
```diff
@@ -357,6 +370,32 @@ def save(self, *args, **kwargs):
...
+    class ScheduledDeactivate(ModalMixin, OrgObjPermsMixin, SmartDeleteView):
+
+        # we create a frontend "delete" ux, but we actually just do a backend update
+        default_template = "broadcast_scheduled_deactivate.haml"
+        cancel_url = "id@msgs.broadcast_scheduled_read"
+        success_url = "@msgs.broadcast_scheduled"
+        fields = ("id",)
+        submit_button_name = _("Delete")
+        # custom object name because broadcasts don't have a name field
+        object_name = ""
```
Comment (rowanseymour) [inline temba/msgs/views.py:382]: remove

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/msgs/views.py [hunk 5]
```diff
@@ -357,6 +370,32 @@ def save(self, *args, **kwargs):
...
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+            context["object_name"] = truncate(self.object.get_text(), 15)
+            return context
+
+        def post(self, request, *args, **kwargs):
+            broadcast = self.get_object()
+            # we don't actually delete, we just deactivate and save
+            broadcast.is_active = False
+            broadcast.save()
```
Comment (rowanseymour) [inline temba/msgs/views.py:393]: Good practice to use `update_fields` except where there are too many fields, so...

```
broadcast.save(update_fields=("is_active",))
```

File: templates/msgs/broadcast_scheduled_delete.haml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+-extends "smartmin/delete_confirm.haml"
+-load smartmin sms temba i18n humanize
+
+-block modal
+  -block delete-message
+    -blocktrans trimmed
+      You are about to delete this
+      %span.font-normal
```
Comment (rowanseymour) [inline templates/msgs/broadcast_scheduled_delete.haml:8]: don't need this formatting.. just..

```
You are about to delete this scheduled message. There is no way to undo this. Are you sure?
```

File: temba/msgs/views.py [hunk 7]
```diff
@@ -357,6 +369,29 @@ def save(self, *args, **kwargs):
             broadcast.save()
             return broadcast
 
+    class ScheduledDelete(ModalMixin, OrgObjPermsMixin, SmartDeleteView):
+
+        # we create a frontend "delete" ux, but we actually just do a backend update
```
Comment (rowanseymour) [inline temba/msgs/views.py:374]: don't need this comment - it's generally how deletes are done

File: temba/tests/crudl.py [hunk 8]
```diff
@@ -181,7 +183,10 @@ def as_user(user, allowed):
             else:
                 checks = [LoginRedirect()]
 
-            return self.requestView(url, user, checks=checks)
+            if as_modal:
+                return self.requestView(url, user, checks=checks, HTTP_X_PJAX=True)
```
Comment (rowanseymour) [inline temba/tests/crudl.py:187]: is this needed?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/msgs/models.py [hunk 9]
```diff
@@ -1008,6 +1008,7 @@ def get_squash_query(cls, distinct_set):
 
         return sql, (distinct_set.org_id, distinct_set.label_type, distinct_set.is_archived) * 2
 
+    # todo figure out if this is how to update the scheduled messages count in the left side nav bar
```
Comment (rowanseymour) [inline temba/msgs/models.py:1011]: clean up

File: temba/msgs/views.py [hunk 10]
```diff
@@ -357,6 +369,29 @@ def save(self, *args, **kwargs):
...
+
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+            return context
+
+        def post(self, request, *args, **kwargs):
+            broadcast = self.get_object()
+            # we don't actually delete, we just deactivate and save
+            broadcast.is_active = False
+            broadcast.save(update_fields=("is_active",))
```
Comment (norkans7) [inline temba/msgs/views.py:389]: I guess we need to update `modified_on` too, probably `modified_by` as well

File: temba/tests/crudl.py (hunk 8 above)
Comment (susanm74) [reply temba/tests/crudl.py:187]: @rowanseymour yeah, @ericnewcomer and i did some troubleshooting on this, and my understanding is that for tests, the view is not returned as a modal by default, so we needed to add this for tests that need the view to be returned as a modal

**Pull request merged**